### DISCUSSION TITLE
Feature Request |trigger action on "Pull Request Approved"

### DISCUSSION THREAD

**Discussion by alexhraber** [community/community#25372] [Actions]
<p>I have a workflow that I have completely built out, only to find out that the actions api doesn’t support “pull-request: approved” (or some variant).</p>
<p>Based on this thread, it seems this is a feature many folks want: <a href="https://github.community/t/github-actions-manual-trigger-approvals/16233">https://github.community/t5/GitHub-Actions/GitHub-Actions-Manual-Trigger-Approvals/td-p/31504/highlight/false</a></p>
<p>None of the current Acivity Types are practical for my use case, and I’m suprised the “approved” activity type is not already supported.</p>
<p>My workflow:</p>
<p>Submit PR -&gt; build triggers -&gt; upon completion a reviewer approves -&gt; upon approval a new action fires off -&gt; merge.</p>
<p>This is very useful for many things, and makes automation simpler. For example – I may want to notify my team in slack that I am merging… I obviously wouldn’t notify before the PR has been aproved.</p>
<p>Well my use case is quite a bit more complex than updating slack – but that is the basic idea. A PR has been approved, and that is a basic GitHub activity that is essential to every developer’s workflow.</p>
<p>Nothing else in the list below will work.</p>
<p>I don’t want our devs typing some magical comment to get things to work, with special regex cases to match every possibility. Same for labels etc.</p>
<p>And the principal idea is “automation”. There are modules to update comments and labels – but what triggers that automatically on approval (which is when we need the action to fire)? The ‘approval action type’ would.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/9b3d14980e244bdd79aada5a68489290d6326ff4.png" data-download-href="/uploads/short-url/m9iS4CuTjy5NN7c0xORaKI6Lz24.png?dl=1" title="Screen Shot 2020-05-13 at 1.55.52 AM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/29316054/181090668-790243da-6b79-4c06-85cc-ffceaa158f73.png" alt="Screen Shot 2020-05-13 at 1.55.52 AM.png" data-base62-sha1="m9iS4CuTjy5NN7c0xORaKI6Lz24" width="690" height="333" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/9b3d14980e244bdd79aada5a68489290d6326ff4_2_690x333.png, https://aws1.discourse-cdn.com/github/original/1X/9b3d14980e244bdd79aada5a68489290d6326ff4.png 1.5x, https://aws1.discourse-cdn.com/github/original/1X/9b3d14980e244bdd79aada5a68489290d6326ff4.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/9b3d14980e244bdd79aada5a68489290d6326ff4_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-05-13 at 1.55.52 AM.png</span><span class="informations">736×356 19.5 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div> <div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/9534dab2212fc075a443dd7593f40262be80fb37.png" data-download-href="/uploads/short-url/lhWnJKOkqn9RCANfgZIEEgzhCoD.png?dl=1" title="Screen Shot 2020-05-13 at 1.56.12 AM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/29316054/181090669-e063fa23-13cc-4d6c-85f0-c8a530d38ba9.png" alt="Screen Shot 2020-05-13 at 1.56.12 AM.png" data-base62-sha1="lhWnJKOkqn9RCANfgZIEEgzhCoD" width="690" height="340" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/9534dab2212fc075a443dd7593f40262be80fb37_2_690x340.png, https://aws1.discourse-cdn.com/github/original/1X/9534dab2212fc075a443dd7593f40262be80fb37.png 1.5x, https://aws1.discourse-cdn.com/github/original/1X/9534dab2212fc075a443dd7593f40262be80fb37.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/9534dab2212fc075a443dd7593f40262be80fb37_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-05-13 at 1.56.12 AM.png</span><span class="informations">746×368 20.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div> <div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/1X/921d1674a851012e619650ebf004e67540515a27.png" data-download-href="/uploads/short-url/kQA1UwmL2EEJXZLzzfpCT8iTl6T.png?dl=1" title="Screen Shot 2020-05-13 at 1.57.38 AM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/29316054/181090670-8e66fc19-07c4-4248-9d6c-0080d2890aa7.png" alt="Screen Shot 2020-05-13 at 1.57.38 AM.png" data-base62-sha1="kQA1UwmL2EEJXZLzzfpCT8iTl6T" width="306" height="500" srcset="https://aws1.discourse-cdn.com/github/optimized/1X/921d1674a851012e619650ebf004e67540515a27_2_306x500.png, https://aws1.discourse-cdn.com/github/optimized/1X/921d1674a851012e619650ebf004e67540515a27_2_459x750.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/1X/921d1674a851012e619650ebf004e67540515a27_2_612x1000.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/1X/921d1674a851012e619650ebf004e67540515a27_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2020-05-13 at 1.57.38 AM.png</span><span class="informations">772×1258 50 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>cc <a class="mention" href="/chrispat">@chrispat</a> <a class="mention" href="/github-staff">@github-staff</a> </p>

**Comment by alexhraber** [comment]
<p>A native solution was found:<br>
<a href="https://github.community/t/github-actions-manual-trigger-approvals/16233/64">https://github.community/t5/GitHub-Actions/GitHub-Actions-Manual-Trigger-Approvals/m-p/58137/highlight/true#M10446</a></p>
<p>Having this a first class citizen is still recommended.</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/everops-alex">@everops-alex</a> ,</p>
<p>Thanks for your sharing. The native solution you provided works perfectly(trigger with ‘pull_request_review: submitted’ event and check ‘github.event.review.state’ value if it’s ‘approved’)! </p>
<p>And it’s recommended to check the value in job level within if expression, it will skip unneccessary jobs to be started.</p>
<pre><code>on:
  pull_request_review:
    types: [submitted]

jobs:
  build:
    if: github.event.review.state == 'approved'
    runs-on: ubuntu-latest
steps: ...
</code></pre>
<p>Regarding the first class citizen feature_request, please raise a ticket <a href="https://support.github.com/contact/feedback?contact%5Bcategory%5D=actions" rel="nofollow noopener">here</a> according to the <a href="https://help.github.com/en/actions/getting-started-with-github-actions/about-github-actions#contacting-support" rel="nofollow noopener">policy</a>, github product manager will take a review and help to confirm.</p>
<p>Cheers!</p>

**Reply by FrostDNA** [reply]
Hello @weide-zhou,

this solution has one problem. It will trigger and skip workflow everytime reviewer submit review and everytime author of PR replies to review comment. This causes following "noise":
![image](https://user-images.githubusercontent.com/104769849/198322850-58b9854b-5464-42b0-a311-81ee74736aa1.png)

Why GHA cannot be configured like this:
```
on:
  pull_request: [approved]
jobs:
build:
runs-on: ubuntu-latest
```
????

**Reply by niall-shaw** [reply]
@weide-zhou - please can you provide an insight as to whether the team is looking into this, as the noise is really offputting

**Reply by ADTC** [reply]
@FrostDNA @niall-shaw try using [delete-workflow-runs v2](https://github.com/marketplace/actions/delete-workflow-runs) with the following config example to clean up the skipped runs. It runs once a day. You can also run it manually by going to the _Actions_ tab, opening it and clicking _Run workflow_ (any branch).

[See Gist for example.](https://gist.github.com/ADTC/ed42e1a9fdb3537b3bb0ad2fb83016c0) Uncomment the `dry_run` line to test it.

# Edit: To clarify here, I'm NOT trying to solve the original problem.
> I'm just helping people by sharing a workaround. If this really bothers you so much, you can set the cron job to run every hour, every 10 minutes, or every minute - effectively you will not see much of it. But asking me why I'm not implementing the option, then 👎ing me for it is like asking some random person on the street why he didn't pass a bill into law in the government, then beating him up for it. **It's hateful bigotry: negative, dismissive, thankless, rude, idiotic and moronic.**
>
> And for fk's sake, most of you are using GitHub for FREE. Do you think you're entitled with a right to demand they do what you want?
>
> # Be nice.

**Reply by FrostDNA** [reply]
Hello @ADTC, this solution is solving the consequence not a root cause of whole problem. Why it's a problem to implement one more option: pull_request: [approved]???

**Reply by ADTC** [reply]
> Hello @ADTC, this solution is solving the consequence not a root cause of whole problem. Why it's a problem to implement one more option: pull_request: [approved]???

@FrostDNA because I don't work for GitHub. That's the problem. F👎 me for trying to help you, right?

**Reply by FrostDNA** [reply]
Oh, sorry :) I thought you work there. My apologies :)

**Comment by ollisala** [comment]
<p>This workaround seems not to work if you want have multiple different workflows. I’m deploying to two environments with different yml-files from branches test and master. In following situation it bugs:</p>
<p>If pull request is done for example against test-branch and someone reviews it, deployment starts also towards another environment (which is integrated to master branch) although in that yml-file there is no mention about test-branch.</p>

**Reply by amrod** [reply]
You can add a branch test to the `if:` statement on the job to ensure the PR was opened against the branch you are expecting.

**Reply by grndvl1** [reply]
This still should be a parent type...not in an if and be able to be used in the "on" field

**Comment by pantelis-karamolegkos** [comment]
This would be very helpful because adding the custom check on jobs `if: github.event.review.state == 'approved'` makes the Actions tab noisy

**Reply by FrostDNA** [reply]
So true look at my comment https://github.com/orgs/community/discussions/25372#discussioncomment-3982987

**Reply by ADTC** [reply]
@pantelis-karamolegkos See my comment for a possible solution to the clean up the noise: https://github.com/orgs/community/discussions/25372#discussioncomment-5432089

**Comment by tassilo-posegga** [comment]
Adding to this: It looks like concurrent jobs are canceled if people leave a review which seems to trigger another run. Following scenario:

- PR gets approved - our ling lasting UI test workflow kicks off
- 10 min into the workflow someone else leaves a comment on the PR - the previous jobs get canceled

```
concurrency:
  group: gherkinTests-${{ github.ref }}
  cancel-in-progress: true
```

I assume a workaround would be to also include the state into the concurrency group

I am looking for a STABLE solution to trigger all our checks once the LAST NECESSARY approval was added to a PR. In our case that would be the second approval
